Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp

```diff
@@ -1010,6 +1010,15 @@ void CachedResourceStreamingClient::responseReceived(PlatformMediaResource&, con
 
     // length will be zero (unknown) if no Content-Length is provided or the response is compressed with Content-Encoding.
     uint64_t length = !response.httpHeaderFields().contains(HTTPHeaderName::ContentEncoding) ? response.expectedContentLength() : 0;
+
+    // But in some cases, Content-Encoding: chunked can specify the total length. Use it.
+    if (!length && response.httpHeaderFields().contains(HTTPHeaderName::TransferEncoding)
+        && (response.httpHeaderField(HTTPHeaderName::TransferEncoding) == String::fromUTF8("chunked"))) {
+        auto contentLength = parseInteger<uint64_t>(response.httpHeaderField(HTTPHeaderName::ContentLength));
```

Comment:
this can be moved to the `if` and no need to call `has_value()` i think?

Reply:
The has_value() is there because the webpage can have the Content-Length header, but a malformed value that may not be parseable. That's why parseInteger() returns an std::optional. In case the optional is empty... I really prefer to ignore whatever default (or even undefined?) value the optional may return in that case.

Reply:
`has_value` and `bool operator` are equivalent, fwiw. https://en.cppreference.com/w/cpp/utility/optional/operator_bool